.github/workflows/ci.yml

```diff
@@ -2,145 +2,186 @@ name: Continuous Integration
 
 on: [push, pull_request, workflow_dispatch]
 
+env:
+  # TEST_ALL_DOCKER_IMAGE: 'parsec-service-test-all'
+  TEST_ALL_DOCKER_IMAGE: 'ghcr.io/parallaxsecond/parsec-service-test-all'
+
 jobs:
+  build-and-export-test-all-docker:
+    runs-on: ubuntu-latest
+    # If TEST_ALL_DOCKER_IMAGE is 'parsec-service-test-all' or any local image,
+    # the following condition must evaluate true to execute this job
+    # Else it must evaluate false to NOT execute this job
+    # Unfortunately, env.TEST_ALL_DOCKER_IMAGE cannot be used here as the `env` context is not recognize at this level.
+    if: ${{ false }} # env.TEST_ALL_DOCKER_IMAGE == 'parsec-service-test-all'
+    steps:
+    - uses: actions/checkout@v3
+    - name: Build the docker container
+      run: pushd e2e_tests/docker_image && docker build -t parsec-service-test-all -f parsec-service-test-all.Dockerfile . && popd
+    - name: Export the docker container
+      run: docker save parsec-service-test-all > /tmp/parsec-service-test-all.tar
+    - name: Upload artifact
+      uses: actions/upload-artifact@v3
+      with:
+        name: parsec-service-test-all
+        path: /tmp/parsec-service-test-all.tar
+
   all-providers:
     name: Various tests targeting a Parsec image with all providers included
     runs-on: ubuntu-latest
+    if: ${{ always() }}
+    needs: [build-and-export-test-all-docker]
     steps:
       - uses: actions/checkout@v2
-      # Use the following step when updating the `parsec-service-test-all` image
-      # - name: Build the container
-      #   run: pushd e2e_tests/docker_image && docker build -t parsec-service-test-all -f parsec-service-test-all.Dockerfile . && popd
+      - name: Load Docker
+        uses: ./.github/actions/load_docker
+        if: ${{ env.TEST_ALL_DOCKER_IMAGE == 'parsec-service-test-all' }}
+        with:
+          image-name: "${{ env.TEST_ALL_DOCKER_IMAGE }}"
+          image-path: "/tmp"
       - name: Run the container to execute the test script
-        run:
-          docker run -v $(pwd):/tmp/parsec -w /tmp/parsec
-          ghcr.io/parallaxsecond/parsec-service-test-all /tmp/parsec/ci.sh all
-      # When running the container built on the CI
-      # run: docker run -v $(pwd):/tmp/parsec -w /tmp/parsec -t parsec-service-test-all /tmp/parsec/ci.sh all
+        run: docker run -v $(pwd):/tmp/parsec -w /tmp/parsec -t ${{ env.TEST_ALL_DOCKER_IMAGE }} /tmp/parsec/ci.sh all
 
   build-all-providers:
     name: Cargo check all-providers (current Rust stable & old compiler)
     runs-on: ubuntu-latest
+    if: ${{ always() }}
```

Comment:
Yes, otherwise, this job won't be executed if the `build-and-export-test-all-docker` is skipped.
The only reason for the `need` list is to preserve order.